aspnet/fundamentals/routing.rst

```diff
@@ -1,114 +1,298 @@
+===================== ====================================================
+/package/create/3     Hello! Route values: [operation, create], [id, 3]
+/package/track/-3     Hello! Route values: [operation, track], [id, -3]
+/package/track/-3/    Hello! Route values: [operation, track], [id, -3]
+/package/track/       <Fall through, no match>
+GET /hello/Joe        Hi, Joe!
+POST /hello/Joe       <Fall through, matches HTTP GET only>
+GET /hello/Joe/Smith  <Fall through, no match>
+===================== ====================================================
 
-The ``defaultHandler`` route handler is used as the default for the ``RouteBuilder``. Calls to ``MapRoute`` will use this handler by default. A second handler is configured within the ``HelloRouter`` instance added by the ``AddHelloRoute`` extension method. This extension methods adds a new ``Route`` to the ``RouteBuilder``, passing in an instance of ``IRouter``, a template string, and an ``IInlineConstraintResolver`` (which is responsible for enforcing any route constraints specified):
+If you are configuring a single route, call ``app.UseRouter`` passing in an ``IRouter`` instance. You won't need to call ``RouteBuilder``.
 
-.. literalinclude:: routing/sample/RoutingSample/HelloExtensions.cs
-  :language: c#
-  :lines: 9-17
-  :dedent: 8
-  :emphasize-lines: 5
+The framework provides a set of extension methods for creating routes such as:
 
-``HelloRouter`` is a custom ``IRouter`` implementation. ``AddHelloRoute`` adds an instance of this router to the ``RouteBuilder`` using a template string, "hello/{name:alpha}". This template will only match requests of the form "hello/{name}" where `name` is constrained to be alphabetical. Matching requests will be handled by ``HelloRouter`` (which implements the ``IRouter`` interface), which responds to requests with a simple greeting.
+- :dn:method:`~Microsoft.AspNetCore.Builder.MapRouteRouteBuilderExtensions.MapRoute`
+- :dn:method:`~Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapGet`
+- :dn:method:`~Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapPost`
+- :dn:method:`~Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapPut`
+- :dn:method:`~Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapRoute`
+- :dn:method:`~Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapVerb`
 
-.. literalinclude:: routing/sample/RoutingSample/HelloRouter.cs
-  :emphasize-lines: 8,20-23
+Some of these methods such as ``MapGet`` require a :dn:delegate:`~Microsoft.AspNetCore.Http.RequestDelegate` to be provided. The ``RequestDelegate`` will be used as the *route handler* when the route matches. Other methods in this family allow configuring a middleware pipeline which will be used as the route handler. If the *Map* method doesn't accept a handler, such as ``MapRoute``, then it will use the :dn:prop:`~Microsoft.AspNetCore.Routing.IRouteBuilder.DefaultHandler`.
```

Comment:
This would be a good place to explain about the `Map[Verb]` methods only matching a particular HTTP method. We could also explain here that they use a constraint to achieve this, since we've already explained constraints